src/reformat_libsharpyuv.c

```diff
@@ -16,22 +16,50 @@ avifResult avifImageRGBToYUVLibSharpYUV(avifImage * image, const avifRGBImage *
-                         rgb->height,
-                         &matrix)) {
+#if SHARPYUV_VERSION >= SHARPYUV_MAKE_VERSION(0, 4, 0)
+    SharpYuvOptions options;
+    SharpYuvOptionsInit(&matrix, &options);
+    if (image->transferCharacteristics == AVIF_TRANSFER_CHARACTERISTICS_UNSPECIFIED) {
+        // Set to sRGB for backward compatibility.
+        options.transfer_type = kSharpYuvTransferFunctionSrgb;
+    } else {
+        options.transfer_type = (SharpYuvTransferFunctionType)image->transferCharacteristics;
+    }
+    const int sharpyuvRes = SharpYuvConvertWithOptions(&rgb->pixels[state->rgbOffsetBytesR],
+                                                       &rgb->pixels[state->rgbOffsetBytesG],
+                                                       &rgb->pixels[state->rgbOffsetBytesB],
+                                                       state->rgbPixelBytes,
+                                                       rgb->rowBytes,
+                                                       rgb->depth,
+                                                       image->yuvPlanes[AVIF_CHAN_Y],
+                                                       image->yuvRowBytes[AVIF_CHAN_Y],
+                                                       image->yuvPlanes[AVIF_CHAN_U],
+                                                       image->yuvRowBytes[AVIF_CHAN_U],
+                                                       image->yuvPlanes[AVIF_CHAN_V],
+                                                       image->yuvRowBytes[AVIF_CHAN_V],
+                                                       image->depth,
+                                                       rgb->width,
+                                                       rgb->height,
+                                                       &options);
+#else
+    const int sharpyuvRes = SharpYuvConvert(&rgb->pixels[state->rgbOffsetBytesR],
+                                            &rgb->pixels[state->rgbOffsetBytesG],
+                                            &rgb->pixels[state->rgbOffsetBytesB],
+                                            state->rgbPixelBytes,
+                                            rgb->rowBytes,
+                                            rgb->depth,
+                                            image->yuvPlanes[AVIF_CHAN_Y],
+                                            image->yuvRowBytes[AVIF_CHAN_Y],
+                                            image->yuvPlanes[AVIF_CHAN_U],
+                                            image->yuvRowBytes[AVIF_CHAN_U],
+                                            image->yuvPlanes[AVIF_CHAN_V],
+                                            image->yuvRowBytes[AVIF_CHAN_V],
+                                            image->depth,
+                                            rgb->width,
+                                            rgb->height,
+                                            &matrix);
+#endif
```

Comment:
I don't how common it is in the codebase to annotate the #endif with a comment (`// SHARPYUV_VERSION >= SHARPYUV_MAKE_VERSION(0, 4, 0)`)

Reply:
About a fourth of the #endif. Then again, we have a nested one here. Done